src/provider/utils/SemVer.ts

```diff
@@ -0,0 +1,114 @@
+export enum SemVerType {
+    VALID = 'VALID',
+    INVALID = 'INVALID',
+    UNKNOWN = 'UNKNOWN'
+}
+
+export enum Operator {
+    LESS_THAN = '<',
+    GREATER_THAN = '>',
+    LESS_THAN_OR_EQUAL = '<=',
+    GREATER_THAN_OR_EQUAL = '>='
+}
+
+/**
+ * Util for comparing semantic version numbers.
+ *
+ * NOTE: Only parses the major, minor, patch and pre-release parts of the semver string - build metadata is ignored.
+ *
+ * Spec: https://semver.org/
+ */
+export class SemVer {
+    private static readonly REGEX_SEMVER: RegExp = /(\d+)\.(\d+)\.(\d+)(?:-([^-+]+))?/;
+    private static readonly REGEX_INTEGER: RegExp = /^\d+$/;
+    private static readonly CACHE: {[key: string]: SemVer} = {};
+
+    public static compare(a: string | SemVer, operator: Operator, b: string | SemVer): boolean {
```

Comment:
why the union type? Don't we just want SemVers, and not strings?

Reply:
As an "ease of use" thing, will implicitly create a `SemVer` object for any incoming strings. This also couples with the caching, which is less effective if done outside of the util.